src/test/java/org/sagebionetworks/bridge/exporter/integration/TestUtils.java

```diff
@@ -61,10 +98,119 @@ public static Table getDdbTable(Config bridgeConfig, DynamoDB ddbClient, String
+            UploadFieldDefinition largeTextFieldDef = new UploadFieldDefinition()
+                    .name(LARGE_TEXT_ATTACHMENT_FIELD_NAME).type(UploadFieldType.LARGE_TEXT_ATTACHMENT);
+            largeTextAttachmentTestSchema = new UploadSchema().schemaId(LARGE_TEXT_ATTACHMENT_SCHEMA_ID)
+                    .revision(LARGE_TEXT_ATTACHMENT_SCHEMA_REV).name("Large Text Attachment Test")
+                    .schemaType(UploadSchemaType.IOS_DATA).addFieldDefinitionsItem(largeTextFieldDef);
+            uploadSchemasApi.createUploadSchema(largeTextAttachmentTestSchema).execute();
+        }
+    }
+
     public static SqsHelper getSqsHelper(AWSCredentialsProvider awsCredentialsProvider) {
         SqsHelper sqsHelper = new SqsHelper();
         //noinspection deprecation
         sqsHelper.setSqsClient(new AmazonSQSClient(awsCredentialsProvider));
         return sqsHelper;
     }
+
+    public static UploadValidationStatus upload(Config config, S3Helper s3Helper, TestUserHelper.TestUser user)
+            throws CertificateEncodingException, CMSException, IOException {
+        // Create file contents.
+        String infoJsonText = "{\n" +
+                "   \"appVersion\":\"version 1.0.0, build 1\",\n" +
+                "   \"createdOn\":\"" + DateTime.now() + "\",\n" +
+                "   \"format\":\"v2_generic\",\n" +
+                "   \"item\":\"" + LARGE_TEXT_ATTACHMENT_SCHEMA_ID + "\",\n" +
+                "   \"phoneInfo\":\"Worker Integ Tests\",\n" +
+                "   \"schemaRevision\":" + LARGE_TEXT_ATTACHMENT_SCHEMA_REV + "\n" +
+                "}";
+
+        String attachmentText = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus orci aliquam.";
+
+        Map<String, String> fileMap = ImmutableMap.<String, String>builder()
+                .put("info.json", infoJsonText)
+                .put(LARGE_TEXT_ATTACHMENT_FIELD_NAME, attachmentText)
+                .build();
+
+        // Zip.
+        byte[] zippedBytes;
+        try (final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                final ZipOutputStream zos = new ZipOutputStream(baos)) {
+            for (Map.Entry<String, String> oneData : fileMap.entrySet()) {
+                ZipEntry zipEntry = new ZipEntry(oneData.getKey());
+                zos.putNextEntry(zipEntry);
+                zos.write(oneData.getValue().getBytes());
+                zos.closeEntry();
+            }
+            zos.flush();
+            zippedBytes = baos.toByteArray();
+        }
+
+        // Encrypt. For whatever reason, the encryptor requires both the public and the private key.
+        // TODO: Fix the encryptor to require only one of the keys, so that the integ tests don't need to download the
+        // private key.
+        String certPem = s3Helper.readS3FileAsString(config.get("upload.cms.cert.bucket"), "api.pem");
+        X509Certificate cert = PemUtils.loadCertificateFromPem(certPem);
```

Comment:
This class will encrypt using only the public cert: https://github.com/Sage-Bionetworks/BridgeDataUploadUtils/blob/develop/src/main/java/org/sagebionetworks/bridge/data/StudyUploadEncryptor.java